streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```
@@ -0,0 +1,468 @@
+    private final StreamsConfig config;
+    private final File stateDir; // default: null
+
+    // settable record metadata ================================================
+    private String topic;
+    private Integer partition;
+    private Long offset;
+    private Long timestamp;
+
+    // mocks ================================================
+    private final Map<String, StateStore> stateStores = new HashMap<>();
+
+
+    /**
+     * A simple class for holding captured punctuators, along with their scheduling information.
+     */
+    public static class CapturedPunctuator {
+        private final long intervalMs;
+        private final PunctuationType type;
+        private final Punctuator punctuator;
+
+        // mutable:
+        private boolean cancelled = false;
+
+        private CapturedPunctuator(final long intervalMs, final PunctuationType type, final Punctuator punctuator) {
+            this.intervalMs = intervalMs;
+            this.type = type;
+            this.punctuator = punctuator;
+        }
+
+        public long getIntervalMs() {
+            return intervalMs;
+        }
+
+        public PunctuationType getType() {
+            return type;
+        }
+
+        public Punctuator getPunctuator() {
+            return punctuator;
+        }
+
+        public void cancel() {
+            this.cancelled = true;
+        }
+
+        public boolean cancelled() {
+            return cancelled;
+        }
+    }
+
+    private final List<CapturedPunctuator> punctuators = new LinkedList<>();
+
+    private class CapturedForward {
+        /*Nullable*/ private final Integer childIndex;
+        /*Nullable*/ private final String childName;
+        private final KeyValue kv;
+
+        private CapturedForward(final KeyValue kv) {
+            this.childIndex = null;
+            this.childName = null;
+            this.kv = kv;
+        }
+
+        private CapturedForward(final Integer childIndex, final KeyValue kv) {
```

If we are already deprecating the API with childIndex, should we simply throw an unsupported exception in the deprecate function and remove this field here as well? WDOT?